src/components/TitleBar/DownloadBtn.tsx

```diff
@@ -10,6 +11,14 @@ import { Colors } from '../../constants';
 export interface DownloadBtnProps {
   url: string;
   filename: string;
+  taskNames?: string[];
+}
+
+export interface DownloadBtnState {
+  url: string;
+  filename: string;
+  modalOpen: boolean;
+  modal: JSX.Element;
```

Comment:
We shouldn't be storing JSX elements in the state. Rather we should store that items props in the state and pass that item the props from the state at render time.